RELEASE.md

```diff
@@ -41,58 +41,68 @@ with `make upgrade-npm-deps`
+- Update the `package.json` files for all packages with the corresponding version:
+  ```bash
+  make bump-version
+  ```
+- Push the branch to github and create a pull request with the release branch as the base. This gives others the opportunity to chime in on the release,
+  in general, and on the addition to the changelog, in particular. 
+  - It's also helpful to drop a link to the release PR in [#perses-dev](https://matrix.to/#/#perses-dev:matrix.org) to get extra visibility.
+- Address any necessary feedback.
+- Once the pull request is approved, merge the it into the release branch.
 
 ### 2. Draft the new release
 
-Tag the new release via the following commands:
-
-```bash
-git checkout release/v<version_to_be_replaced>
-make tag
-git push origin v<version_to_be_replaced>
-```
+- Pull down the latest updates to the release branch on your local machine to ensure you have the updates from the previous step.
+- Tag the new release via the following commands:
+  ```bash
+  git checkout release/v<version_to_be_replaced>
+  make tag
+  git push origin v<version_to_be_replaced>
+  ```
 
 Signing a tag with a GPG key is appreciated, but in case you can't add a GPG key to your Github account using the
 following [procedure](https://docs.github.com/en/authentication/managing-commit-signature-verification), you can replace
 the `-s` flag by `-a` flag of the git tag command to only annotate the tag without signing.
 
-Once a tag is created, the release process through the GitHub Actions will be triggered for this tag.
+Once a tag is created, an automated release process for this tag is triggered via Github Actions. This automated process includes:
+- Publishing new versions of the UI libraries to npm.
+- Building new go binaries and docker images.
+- Creating a new Github release that uses the changelog as the release notes and provides tarballs with the latest go binaries & docker images.
+
+Please verify that the Github Actions complete successfully. Once they are completed, check that everything looks good in the new Github Release. If you realize we need to adjust something in the release notes, you can edit it directly in the Github UI.
+
+### 3. Merge the release into `main`
+
+It can be helpful to leave the release branch up for a little while in case we need to create a patch release to address bugs or minor issues with the release you just made. 
 
-The Github releases will be created automatically by the GitHub Action triggered. Hopefully you will find the accurate
-changelog there. It won't hurt if you check quickly if it does not miss anything there.
+Once the release branch is no longer needed, you should open a new PR based on `main` to merge those changes. When this PR is approved, merge it into `main` using the "merge pull request" option (instead of "squash and merge," which can have issues with release branches).
```

Comment:
@sjcobb I think you were the person who mentioned the recommendation to use "merge" (instead of "squash and merge"). Do you have any additional context that should get added here?

Reply:
the branch `release vX.Y` is containing the commit tagged. So if you are going to squash this branch in `main`, then this commit will disappear in the git history and then the tag is not attached to any known commit.

That's why it's import to use `merge` instead because it will preserve the commits made on the release branch

Reply:
I found the original discussion on Matrix when this came up [here](https://matrix.to/#/!XNmgYIAndZOkEvQAbb:matrix.org/$jLLD7HCHNbdxMDZQJOHNqUt0ku5jFhD2QSkLfWTs1DE?via=matrix.org&via=herail.net). You can see what happened too if you check some older releases like [v0.7.1](https://github.com/perses/perses/tree/v0.7.1):

<img width="1038" alt="image" src="https://user-images.githubusercontent.com/9369625/233099967-11c17a4e-b744-4d26-a4ff-e6f3f4bdab3f.png">

At one point, I looked through all the settings in GitHub to see if we could automate this somehow and couldn't find a way, it gets tricky too since it always defaults to however you merged your last PR so sometimes it's annoying to remember to change back to 'Squash' the next time since we do want to Squash every other type of PR 😕

Reply:
One unknown for me is, does this only happen on patch releases? The example above is `v0.7.1` where I was merging in multiple commits that got squashed. I'm pretty sure before that release, I did squash some releases that were normal minor release versions and not a patch release, and it worked fine 🤷‍♂️

Reply:
I'm wondering again about the whole minor release branch workflow, it definitely has some upsides about letting development continue on `main` branch and pushing out quick patch releases for bugfixes, but since we have to sync the minor release branch with main anyways before the next minor release can go out, that negates some of the benefits to me. 

This came up with Eunice's breaking changes in our last `v0.28.0` release. I [posted on Matrix](https://matrix.to/#/!XNmgYIAndZOkEvQAbb:matrix.org/$uy-7vPmyIjUtjlxncWV06smk453TS_C3pVo7iTbRLvo?via=matrix.org&via=herail.net) that I wanted to keep the `release/v0.27` open in case we needed to get bugfixes out before we could manually migrate all our dashboards internally (side note: we really need some automated way to do these schema migrations). But then when Eunice went to do the release, we realized that keeping that branch open wasn't an option since we needed to sync the changelog and version changes before she could do her release. 

I've posted about maybe changing to a workflow where we cherrypick just the PRs we need back into an older release branch to do patch releases w/o holding up `main` but the downside there is we might run into this issue where the tags end up in state like `v0.7.1` did (`This commit does not belong to any branch on this repository` error). Any thoughts @juliepagano are there other options we aren't considering? For context, we picked our current process because it works well for the Prometheus project ([ex](https://github.com/prometheus/prometheus/blob/main/RELEASE.md)), but since their releases go out every ~6 weeks and ours are more frequent, the needs are somewhat different

Reply:
Related, I wrote up a concise version of the steps I did to do the v0.27.0 release [here](https://github.com/perses/perses/pull/1098):

<img width="754" alt="image" src="https://user-images.githubusercontent.com/9369625/233106313-9b9399af-69db-4ef7-9c3f-f9899fd03268.png">

Now that we have changelog generation that works really well and GitHub gives us the ability to manually edit release notes after the release goes out to clean up any PRs with 'unknowns', I bet we could automate at least half of the bullet points. What if we could just cut a release by making a `release/v0.x.x` branch and then basically everything automatically happens including the last steps where the release is tagged (assuming a maintainer cut the release branch and we figure out some way to do the GPG key signing in CI)

Reply:
@sjcobb would you be up for moving the thoughts about adjusting the process to a github discussion? I think that will be a slightly better venue for us to work through tradeoffs and how we want to adjust things. I could see it taking a bit to work out something new, so I think there's value in having this PR focus on clearly defining our current process until we land on something new.

Reply:
Yeah agreed! Thanks for updating this by the way 🙏